burotel/indico_burotel/plugin.py

```diff
@@ -35,19 +49,66 @@ def _add_missing_time(field, data, time_):
         data[field] = dts.dump({field: datetime.combine(date, time_)})[field]
 
 
+def _check_update_permissions(booking):
+    if booking.room.attributes.filter(
+        RoomAttribute.name == 'electronic-lock',
+        RoomAttributeAssociation.value == cast('yes', JSONB)
+    ).join(Room.attributes, RoomAttributeAssociation.attribute).first():
+        @after_this_request
+        def _launch_task(response):
+            update_access_permissions.delay(booking)
+            return response
+
+
+def _check_no_parallel_bookings(booking):
+    """Ensure that the user has no other bookings in that interval"""
+    overlapping = query_user_overlapping_bookings(booking).all()
+    if overlapping:
+        raise ExpectedError(_("There is a parallel booking for this person in {}, from {} to {}").format(
+            overlapping[0].room.full_name,
+            format_date(overlapping[0].start_dt),
+            format_date(overlapping[0].end_dt)
+        ))
+
+
+class SettingsForm(IndicoForm):
+    _fieldsets = [
+        ('ADaMS Sync', ['adams_service_url', 'adams_username', 'adams_password', 'cern_identity_provider'])
+    ]
+
+    adams_service_url = URLField(_('Service URL'), [URL(require_tld=False)],
+                                 description=_('The URL of the ADaMS service. You must use the {action}, {room}, '
+                                               '{person_id}, {start_dt} and {end_dt} placeholders.'))
+    adams_username = StringField(_('Username'), validators=[DataRequired()],
+                                 description=_('The username to access the ADaMS web service'))
+    adams_password = IndicoPasswordField(_('Password'), [DataRequired()], toggle=True,
+                                         description=_('The password to access the ADaMS web service'))
+    cern_identity_provider = StringField(_('CERN Identity Provider'), validators=[DataRequired()])
```

Comment:
This could be a `SelectField` populated from `multipass.identity_providers`

Reply:
Jolly good idea!